### PR TITLE
Feature notification models/routes

### PR DESCRIPTION
**Pull request by ghost**
### What this PR does (required):
- Created Model, Routes, and Controllers for Notification feature. The actual model is for a PriceDrop. More models could be added to the file. 

### Screenshots / Videos (required):
- Notification schema:
image
- Added notifications array to user model:
image

- Routes
image

- Controllers
Get Unread:
image
Mark as read:
image
Get all notifications:
image

### Any information needed to test this feature (required):
- the TaskQueue check periodically for price changes. If a price drops, it creates a notification and adds it to the user model that has this product in its list.
- With a login cookie, you can: 
   GET to /notification/:userId    (get all notifications)
   GET to /notification/unread/:userId (get unread notifications)
   PUT to /notification/:id   (mark notification as read, where :id is the notification :id)

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -13,17 +13,26 @@ const { setScraperInterval } = require("./utils/taskQueues");
...
+  cors: {
+    origin: "*",
+  },
+});
+
+io.on("connection", (socket) => {
+  console.log("connected");
 	cors: {
 		origin: "*",
 	},
```
Comment (dproc96) [inline server/app.js]: Should this be removed?

**Review by dproc96 — COMMENTED** [review]
I think this task description was a little unclear, the scraper should be creating the price change notification when it detects a price drop. Can you refactor to accomplish that?

File: server/app.js (hunk 1 above)
Comment (ghost) [reply server/app.js]: Not at all. My mistake.

File: server/controllers/notification.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            productId: priceDropNotification.productId,
+            oldPrice: priceDropNotification.oldPrice,
+            url: priceDropNotification.url,
+            userId: priceDropNotification.userId,
+          },
+        },
+      });
+    } else {
+      res.status(400);
+      throw new Error("Invalid notification data");
```
Comment (bonnieli) [inline server/controllers/notification.js]: throwing an error without a message sent back as a response seems inconsistent with other routes

**Review by bonnieli — APPROVED** [review]

File: server/controllers/notification.js (hunk 2 above)
Comment (ghost) [reply server/controllers/notification.js]: Right. Corrected. Thank you for noting.

**Comment by ghost** [comment]
> I think this task description was a little unclear, the scraper should be creating the price change notification when it detects a price drop. Can you refactor to accomplish that?

Yes. I just updated the PR description with the final working program and added more clear information.

**Pull request merged**